### PR TITLE
fix: Application name mismatch when fiat permissions coming from diff…

### PR DESCRIPTION
**Pull request by sanopsmx**
…erent sources. #5156

This is part of: spinnaker/spinnaker#5156.

Fixed the name mismatch bug in ResourcePrefixPermissionSource.java

Resource(Application) always returns the name in CAPITAL letters.

Permission prefixes added in external sources in fiat-local.yml has the application name in lowercase letters.

Modified the code to check these names with

1)prefix.equalsIgnoreCase(resource.getName());

2) prefixWithoutStar = prefixWithoutStar.toUpperCase();

Modified the ResourcePrefixPermissionSourceSpec.java to fix the testcases.

### CODE REVIEW

File: fiat-roles/src/test/groovy/com/netflix/spinnaker/fiat/providers/ResourcePrefixPermissionSourceSpec.groovy [hunk 1]
```diff
@@ -44,9 +44,9 @@ class ResourcePrefixPermissionSourceSpec extends Specification {
 
         where:
         applicationName     | expectedCreatePermissions
-        'new york'          | ["admins"]
-        'gotham-criminals'  | ["admins", "police"]
-        'gotham-joker'      | ["admins", "police", "batman"]
+        'NEW YORK'          | ["admins"]
+        'GOTHAM-CRIMINALS'  | ["admins", "police"]
```
Comment (cfieber) [inline fiat-roles/src/test/groovy/com/netflix/spinnaker/fiat/providers/ResourcePrefixPermissionSourceSpec.groovy]: can you update these test cases to include a mix of upper and lower case (it looks like your changes should ensure that works)

Comment (sanopsmx) [reply fiat-roles/src/test/groovy/com/netflix/spinnaker/fiat/providers/ResourcePrefixPermissionSourceSpec.groovy]: Added the testcases.

Comment (sanopsmx) [reply fiat-roles/src/test/groovy/com/netflix/spinnaker/fiat/providers/ResourcePrefixPermissionSourceSpec.groovy]: Added the testcases.

**Comment by sanopsmx** [comment]
Added new test cases and waiting for the feedback

**Comment by sanopsmx** [comment]
Added new test cases and waiting for the feedback. Can you please check in the code.

**Comment by sanopsmx** [comment]
Can you please 
review and merge the changes

**Pull request merged**